src/chunk.c

```diff
@@ -294,18 +328,74 @@ chunk_collision_resolve(Hyperspace *hs, Hypercube *cube, Point *p)
 	chunk_scan_ctx_destroy(&scanctx);
 }
 
+static int
+chunk_add_constraints(Chunk *chunk)
+{
+	int			num_added;
+
+	num_added = chunk_constraints_add_dimension_constraints(chunk->constraints,
+															chunk->fd.id,
+															chunk->cube);
+	num_added += chunk_constraints_add_inheritable_constraints(chunk->constraints,
+															   chunk->fd.id,
+													chunk->hypertable_relid);
+
+	return num_added;
+}
+
+static Oid
+chunk_create_table(Chunk *chunk, Hypertable *ht)
+{
+	Catalog    *catalog = catalog_get();
+	Relation	rel;
+	ObjectAddress objaddr;
+	int			sec_ctx;
+	CreateStmt	stmt = {
+		.type = T_CreateStmt,
+		.relation = makeRangeVar(NameStr(chunk->fd.schema_name), NameStr(chunk->fd.table_name), 0),
+		.inhRelations = list_make1(makeRangeVar(NameStr(ht->fd.schema_name), NameStr(ht->fd.table_name), 0)),
+		.tablespacename = hypertable_select_tablespace(ht, chunk),
+	};
+	Oid			uid,
+				saved_uid;
+
+	rel = heap_open(ht->main_table_relid, AccessShareLock);
+
+	if (namestrcmp(&chunk->fd.schema_name, INTERNAL_SCHEMA_NAME) == 0)
+		uid = catalog->owner_uid;
```

Comment:
I am not sure we want anyone to be able to create any type of tables in the internal schema, because users can then do a manual `CREATE TABLE` in that schema, which I don't think they should be able to do by default.

What this logic does is the following:

1. if the chunk is created in the internal schema, we create it with the permissions of the catalog/schema owner. (Still, the chunk's owner will be the owner of the hypertable.)
2. if the chunk is created in a user-specified "associated schema", then we shouldn't use the catalog owner to create the table since that typically implies super-user permissions. If we allow that, anyone can specify someone else's schema in `create_hypertable()` and start creating chunks in it without having actual permissions on the schema to do so. With this logic, the hypertable owner must have permissions to create tables in the associated schema, or else table creation will fail. If the schema doesn't yet exist, the table owner need the proper permissions on the database to create the schema.

Reply:
Can you put a comment explaining this?

Reply:
done